DropboxLike.Domain/Repositories/FileRepository.cs

```diff
@@ -47,127 +47,167 @@ public async Task<S3Response> UploadFileAsync(IFormFile file)
         };
 
         _applicationDbContext.FileModels.Add(fileModel);
-        // TODO: Check if file already exist
         await _applicationDbContext.SaveChangesAsync();
-
-        _response.StatusCode = 200;
-        _response.Message = $"{file.Name} has been uploaded to s3 successfully";
+        
+        return OperationResult<object>.Success(new object(), HttpStatusCode.Created);
       }
     }
-    catch (AmazonS3Exception ex)
+    catch (AmazonS3Exception exception)
     {
-      _response.StatusCode = (int)ex.StatusCode;
-      _response.Message = ex.Message;
+      var message = $"{exception.StatusCode}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message, exception.StatusCode);
     }
-    catch (Exception ex)
+    catch (Exception exception)
     {
-      _response.StatusCode = 500;
-      _response.Message = ex.Message;
+      var message = $"{HttpStatusCode.InternalServerError}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message);
     }
-    return _response;
   }
 
-  public async Task<FileStreamResult> DownloadFileAsync(string fileId)
+  public async Task<OperationResult<File>> DownloadFileAsync(string fileId)
   {
-
-    // var results = await _applicationDbContext.FileModels
-    //     .FirstOrDefaultAsync(x => x.FileKey == fileId);
-
-    //var destinationFolderPath = $"/home/godfreyowidi/Downloads/TestsDowloads";
-
     try
     {
       var file = await _applicationDbContext.FileModels.FindAsync(fileId);
-
       if (file == null)
       {
         throw new FileNotFoundException("File not Found");
       }
 
-      var objectKey = file?.FileKey?.ToString();
-
-      ListObjectsV2Request listRequest = new ListObjectsV2Request
+      var listRequest = new ListObjectsV2Request
       {
         BucketName = _bucketName,
       };
-      ListObjectsV2Response listResponse = await _awsS3Client.ListObjectsV2Async(listRequest);
+      
+      // This code just ensure we retrieve only object names - not all metadata
+      // Caching of the retrieved names for 20 mins - server side caching.
+      var cache = new MemoryCache(new MemoryCacheOptions());
+      if (cache.TryGetValue<List<string>>(_bucketName, out var fileNames))
+      {
+        var fileCache = new File
+        {
+          ObjectNames = fileNames
+        };
+        return OperationResult<File>.Success(fileCache);
+      }
+      fileNames = new List<string>();
+      
+      ListObjectsV2Response listResponse;
+      do
+      {
+        listResponse =  await _awsS3Client.ListObjectsV2Async(listRequest);
+        
+        fileNames.AddRange(listResponse.S3Objects.Select(obj => obj.Key));
+
+        listRequest.ContinuationToken = listResponse.NextContinuationToken;
+      } while (listResponse.IsTruncated);
+
+      var cacheOptions = new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(10));
+
+      cache.Set(_bucketName, fileNames, cacheOptions);
```

Comment:
Since you have only one bucket, you are essentially loading the entire collection of files stored in this bucket whenever this code is evaluated. This does not scale. Ideally, you should be caching on a per-user basis.
Also consider using a service like AWS ElastiCache or Redis for caching instead, although this does complicate your architecture.